### DISCUSSION TITLE
Issue on deployment from Github

### DISCUSSION THREAD

**Discussion by jmarbutt** [hasura/graphql-engine#8915] [General] [answered]
I am getting the following issue on a Github deployment 

```
github-integration failed to complete because error applying metadata: {
  "code": "parse-failed",
  "error": "the key 'definition' was not present",
  "path": "$.args.metadata.query_collections[0]"
}
```

Any ideas why? I have searched all my metadata for any of those terms and don't see anything.

**Comment by jmarbutt** [comment]
I also tried creating a new Hasura Cloud project and created a heroku db and turned on the github integration and got the same thing with a fresh database.

**Comment by meetzaveri** [comment, marked as answer]
Hello @jmarbutt ,

This is related to GH ticket - https://github.com/hasura/graphql-engine/issues/8492

### Synopsis:
Acc. to hasura metadata docs, `definition` and `name` key are required fields for query collection API. Doc link -https://hasura.io/docs/latest/graphql/core/api-reference/metadata-api/query-collections/ 

Can you verify your metadata if they are missing ?

### Workaround
If you are using hasura cloud,  what you could do is
- Import metadata (check if those are missing)
- Update metadata manually by adding those fields (you can also refer above GH issue for workaround)
- And then upload that updated metadata by doing "export metadata" in hasura console.

### Further progress/fix
We believe this should be a CLI error and it should inform that in which file and at which LOC it is so that you can debug easily. We also are working on fix for that. Reference GH ticket - https://github.com/hasura/graphql-engine/issues/8503

I hope this helps. If not, please feel free to comment again and let us know

**Reply by jmarbutt** [reply]
Yes this was missing, I will add it manually. I originally tried the approach of recreating my metadata via the `hasura metadata export` cli command and it did not fix it.
